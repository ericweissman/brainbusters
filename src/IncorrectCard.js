import React, { Component } from 'react';
import './styles/Main.scss';

class IncorrectCard extends Component {
  render() {
    return (
      <div className="incorrect-card">
        <h4>Wrong!</h4>
        <p>{this.props.question}</p>
        <p>Correct Answer: {this.props.correct_answer}</p>
        <p>Your Answer: {this.props.incorrect_answer}</p>
        <button onClick={() => this.props.updateGuessedCards(this.props.id)}>Back to Quiz</button>
      </div>
    )
  }
}

export default IncorrectCard;