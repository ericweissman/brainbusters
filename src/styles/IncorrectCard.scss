.incorrect-card {
  width: 500px;
  border-radius: 15px;
  border: 2px solid $dark;
  margin: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $dark;
  color: whitesmoke;

  h4 {
    font-family: $title;
    color: $red;
    font-size: 40px;
    margin: 5px auto;
  }
  
  p {
    width: 285px;
    text-align: center;
    font-family: $text;
    font-weight: 800;
    margin: 5px auto;
  }

  img {
    height: 175px;
    border: 2px solid $red;
    border-radius: 5px;
  }

  button {
    background-color: $white;
    border-radius: 15px;
    padding: 5px;
    width: 100px;
    margin: 10px;
    &:hover {
        background: linear-gradient(to right, #E100FF, #7F00FF);
        color: $white;
        cursor: pointer;
    }
  }
}

  .incorrect-answer {
    color: $red;
    font-weight: 800;
    font-size: 25px;
  }