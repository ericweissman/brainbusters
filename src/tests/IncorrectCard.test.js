import React from 'react';
import IncorrectCard from '../IncorrectCard.js'
import { shallow } from 'enzyme';

const question = {
  "id": 1,
  "question": "What array prototype method would return a sum of the elements in an array?",
  "answers": [
    ".sort()",
    ".reduce()",
    ".forEach()"
  ],
  "correct_answer": ".reduce()",
  "concept": "prototypes"
}
const correct_answer = "Testing is my favorite hobby."
const incorrect_answer = "Daggob!"
const id = 21
const updateGuessedCardsMock = jest.fn()
const updateStudyListMock = jest.fn()
const showAllCards = true;
const resetMock = jest.fn();

describe('IncorrectCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <IncorrectCard
        question={question}
        correct_answer={correct_answer}
        incorrect_answer={incorrect_answer}
        id={id}
        updateGuessedCards={updateGuessedCardsMock}
        showAllCards={showAllCards}
        updateStudyList={updateStudyListMock}
        reset={resetMock}
      />
    )
  });

  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateguessedcards and update study list on button click', () => {
    wrapper.find('.back-to-quiz-button').simulate('click');
    expect(updateGuessedCardsMock).toBeCalled();
    expect(updateStudyListMock).toBeCalled();
  })

});