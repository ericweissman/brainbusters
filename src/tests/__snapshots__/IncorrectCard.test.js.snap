// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncorrectCard should match snapshot when all data is passed correctly 1`] = `
<div
  className="incorrect-card"
>
  <h4>
    Wrong!
  </h4>
  <img
    src="https://media.giphy.com/media/BmshHM2V8ByjC/giphy.gif"
  />
  <p>
    <Component />
  </p>
  <p
    className="display-correct"
  >
    Correct Answer: 
    Testing is my favorite hobby.
  </p>
  <p
    className="incorrect-answer"
  >
    Your Answer: 
    Daggob!
  </p>
  <button
    className="back-to-quiz-button"
    onClick={[Function]}
  >
    Back to Quiz
  </button>
</div>
`;
